### PR TITLE
[ISSUE #3905] Support bname in protocol for 4.9.x client

### PR DESCRIPTION
**Pull request by drpmma**
## What is the purpose of the change

Backport 5.0 bname protocol for 4.9.x client

## Brief changelog

See [this issue](https://github.com/apache/rocketmq/issues/3905) for more information

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/processor/ReplyMessageProcessor.java [hunk 1]
```diff
@@ -173,6 +173,7 @@ private PushReplyResult pushReplyMessage(final ChannelHandlerContext ctx,
         replyMessageRequestHeader.setProperties(requestHeader.getProperties());
         replyMessageRequestHeader.setReconsumeTimes(requestHeader.getReconsumeTimes());
         replyMessageRequestHeader.setUnitMode(requestHeader.isUnitMode());
+        replyMessageRequestHeader.setBname(requestHeader.getBname());
```
Comment (lizhanhui) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/ReplyMessageProcessor.java:176]: `bname` is arcane and inconsistent with the existing field naming convention.  Why not use brokerName?

Comment (lizhanhui) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/ReplyMessageProcessor.java:176]: `bname` is arcane and inconsistent with the existing field naming convention.  Why not use brokerName?

Comment (lizhimins) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/ReplyMessageProcessor.java:176]: +1

Comment (drpmma) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/ReplyMessageProcessor.java:176]: It's backport from 5.0 branch. The `bname` field is introduced by https://github.com/apache/rocketmq/commit/b6ff649291453d3dc2d702fbba1548d991b610ac.

File: common/src/main/java/org/apache/rocketmq/common/protocol/header/CheckTransactionStateRequestHeader.java [hunk 2]
```diff
@@ -20,18 +20,19 @@
...
-public class CheckTransactionStateRequestHeader implements CommandCustomHeader {
+public class CheckTransactionStateRequestHeader extends RpcRequestHeader {
     @CFNotNull
     private Long tranStateTableOffset;
     @CFNotNull
     private Long commitLogOffset;
     private String msgId;
     private String transactionId;
     private String offsetMsgId;
+    private int queueId;
```
Comment (xdkxlk) [inline common/src/main/java/org/apache/rocketmq/common/protocol/header/CheckTransactionStateRequestHeader.java:36]: The CheckTransactionStateRequestHeader is sent by broker. I think we should change the org.apache.rocketmq.broker.client.net.Broker2Client#checkProducerTransactionState to set the new fileds

Comment (drpmma) [reply common/src/main/java/org/apache/rocketmq/common/protocol/header/CheckTransactionStateRequestHeader.java:36]: It's only client-side modification.

File: common/src/main/java/org/apache/rocketmq/common/protocol/header/EndTransactionRequestHeader.java [hunk 3]
```diff
@@ -128,4 +130,12 @@ public String toString() {
             ", transactionId='" + transactionId + '\'' +
```
[the remark this answers is not in the document]
Comment (drpmma) [reply common/src/main/java/org/apache/rocketmq/common/protocol/header/EndTransactionRequestHeader.java:130]: added

**Pull request merged**